Source/Urho3D/CSharp/Managed/Core/Context.cs

```diff
@@ -106,6 +106,11 @@ public void RegisterFactory(Type type, string category="")
                 throw new InvalidOperationException("This type can not be registered as factory.");
 
             Urho3D_Context_RegisterFactory(getCPtr(this), type.Name, StringHash.Calculate("SwigDirector_" + baseType.Name), category);
+            if (type.IsSubclassOf(typeof(Serializable)))
+            {
+                var serializable = (Serializable)Activator.CreateInstance(type, new object[] { Context.Instance });
+                serializable.RegisterAttributes();
```

Comment:
Not strictly incorrect, but better would be:
```cs
using (var serializable = (Serializable)Activator.CreateInstance(type, new object[] { Context.Instance }))
{
    serializable.RegisterAttributes();
}
```
This essentially calls `serializable.Dispose()` automatically and thus gets rid of an object immediately.